fire/fire_test.py

```diff
@@ -368,10 +368,15 @@ def testBoolParsingLessExpectedCases(self):
         fire.Fire(tc.MixedDefaults,
                   command=['identity', 'True', '10']), (True, 10))
 
```

Comment:
Another thing that may be worthwhile to test is the behavior in the presence of separators / chained function calls.

For example, with these components:

```
def example_component(arg1, kwarg2='default2'):
  def inner_component(kwarg3='default3'):
    return kwarg3
  return inner_component
```

```
def example_component(arg1, *varargs):
  def inner_component(kwarg3='default3'):
    return kwarg3
  return inner_component
```

What is the behavior of these calls?

fire.Fire(example_component, command=['value1', '-', '--kwarg3', 'value3']) == 'value3'
fire.Fire(example_component, command=['value1', '--kwarg3', 'value3'])